awx/main/utils/common.py

```diff
@@ -168,20 +168,25 @@ def memoize_delete(function_name):
     return cache.delete(function_name)
 
 
-@memoize()
-def get_ansible_version():
+def _get_ansible_version(ansible_path):
     '''
     Return Ansible version installed.
+    Ansible path needs to be provided to account for custom virtual environments
     '''
     try:
-        proc = subprocess.Popen(['ansible', '--version'],
+        proc = subprocess.Popen([ansible_path, '--version'],
                                 stdout=subprocess.PIPE)
         result = smart_str(proc.communicate()[0])
         return result.split('\n')[0].replace('ansible', '').strip()
     except Exception:
         return 'unknown'
 
 
+@memoize()
+def get_ansible_version(ansible_path='ansible'):
```

Comment:
This isn't a good practice here.

Since this method is memoized, it shouldn't take any argument. That only begs for bugs. I say this should take no argument, and call `_get_ansible_version('ansible')`

Reply:
Why does this need an argument at all ?

Shouldn't the code that's in [BaseTask.get_ansible_version()](https://github.com/ansible/awx/blob/6306255e43fcd04e8cc16532a303504f5ae0f5c0/awx/main/tasks.py#L692-L697), which does *not* use `self`, be defined at least in this module by default ?